code/modules/surgery/organs/lungs.dm

```diff
@@ -113,10 +160,389 @@
+		// tl;dr the first argument chooses the smaller of miasma_pp/2 or 6(typical max virus symptoms), the second chooses the smaller of miasma_pp or 8(max virus symptom level)
+		// Each argument has a minimum of 1 and rounds to the nearest value. Feel free to change the pp scaling I couldn't decide on good numbers for it.
+		miasma_disease.name = "Unknown"
+		miasma_disease.try_infect(breather)
+	// Miasma side effects
+	switch(miasma_pp)
+		if(0.25 to 5)
+			// At lower pp, give out a little warning
+			breather.clear_mood_event("smell")
+			if(prob(5))
+				to_chat(breather, span_notice("There is an unpleasant smell in the air."))
+		if(5 to 15)
+			//At somewhat higher pp, warning becomes more obvious
+			if(prob(15))
+				to_chat(breather, span_warning("You smell something horribly decayed inside this room."))
+				breather.add_mood_event("smell", /datum/mood_event/disgust/bad_smell)
+		if(15 to 30)
+			//Small chance to vomit. By now, people have internals on anyway
+			if(prob(5))
+				to_chat(breather, span_warning("The stench of rotting carcasses is unbearable!"))
+				breather.add_mood_event("smell", /datum/mood_event/disgust/nauseating_stench)
+				breather.vomit()
+		if(30 to INFINITY)
+			//Higher chance to vomit. Let the horror start
+			if(prob(15))
+				to_chat(breather, span_warning("The stench of rotting carcasses is unbearable!"))
+				breather.add_mood_event("smell", /datum/mood_event/disgust/nauseating_stench)
+				breather.vomit()
+		else
+			breather.clear_mood_event("smell")
+	// In a full miasma atmosphere with 101.34 pKa, about 10 disgust per breath, is pretty low compared to threshholds
+	// Then again, this is a purely hypothetical scenario and hardly reachable
+	breather.adjust_disgust(0.1 * miasma_pp)
+
+/// We're free from the stick, clear out its impacts
+/obj/item/organ/internal/lungs/proc/safe_miasma(mob/living/carbon/breather, datum/gas_mixture/breath, old_miasma_pp)
+	// Clear out moods when immune to miasma, or if there's no miasma at all.
+	breather.clear_mood_event("smell")
+
+/// Causes random euphoria and giggling. Large amounts knock you down
+/obj/item/organ/internal/lungs/proc/too_much_n2o(mob/living/carbon/breather, datum/gas_mixture/breath, n2o_pp, old_n2o_pp)
+	if(n2o_pp < n2o_para_min)
+		// Small amount of N2O, small side-effects.
+		if(n2o_pp <= 0.01)
+			if(old_n2o_pp > 0.01)
+				return BREATH_LOST
+			return
+		// No alert for small amounts, but the mob randomly feels euphoric.
+		if(old_n2o_pp >= n2o_para_min || old_n2o_pp <= 0.01)
+			breather.clear_alert(ALERT_TOO_MUCH_N2O)
+
+		if(prob(20))
+			n2o_euphoria = EUPHORIA_ACTIVE
+			breather.emote(pick("giggle", "laugh"))
+		else
+			n2o_euphoria = EUPHORIA_INACTIVE
+		return
+
+	// More N2O, more severe side-effects. Causes stun/sleep.
+	if(old_n2o_pp < n2o_para_min)
+		breather.throw_alert(ALERT_TOO_MUCH_N2O, /atom/movable/screen/alert/too_much_n2o)
+	n2o_euphoria = EUPHORIA_ACTIVE
+
+	// 60 gives them one second to wake up and run away a bit!
```

Comment:
no need to have this comment anymore because it's no longer correct anyways (did they really not know how deciseconds worked???)

Reply:
lmao